### PR TITLE
Refactor some modules under core umbrella module

### PR DESCRIPTION
**Pull request by varunnvs92**
Current `core` renamed to `sdk-core`

Umbrella core module contains following child modules:
`annotations`
`auth`
`sdk-core`
`aws-core`
`profiles`
`regions`

### CODE REVIEW

**Comment by zoewangg** [comment]
We might need to update the package name in core to `software.amazon.awssdk.sdkcore`

File: pom.xml [hunk 1]
```diff
@@ -73,8 +68,8 @@
         <module>test/service-test-utils</module>
         <module>test/test-utils</module>
         <module>test/codegen-generated-classes-test</module>
-        <module>annotations</module>
         <module>utils</module>
+        <module>core</module>
```
Comment (zoewangg) [inline pom.xml]: nit, can we move `core` in the top and move `utils` before test modules? It's nice to separate non-test modules and test modules.

File: core/pom.xml [hunk 2]
```diff
@@ -1,233 +1,28 @@
...
-  ~ on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
-  ~ express or implied. See the License for the specific language governing
-  ~ permissions and limitations under the License.
-  -->
-
-<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd"
-         xmlns="http://maven.apache.org/POM/4.0.0"
-         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
```
Comment (zoewangg) [inline core/pom.xml]: missing copyright here

**Comment by varunnvs92** [comment]
I thought of it but it felt weird to have awssdk and sdkcore (2 sdks side by side). Also we don't have any source code in core module, so i felt having awssdk.core is fine. I can change it if we strictly follow module name convention in java package name.

File: core/pom.xml (hunk 2 above)
Comment (varunnvs92) [reply core/pom.xml]: What is the criteria for adding copyright to pom? I see some other modules don't have it like auth, aws-core, annotations

Comment (zoewangg) [reply core/pom.xml]: I'd assume we should add it to all files

**Comment by varunnvs92** [comment]
Closing in favor of https://github.com/aws/aws-sdk-java-v2/pull/589

**Pull request closed without merging**